tanner/session_analyzer.py

```diff
@@ -150,3 +136,36 @@ def find_location(ip):
         except geoip2.errors.AddressNotFoundError:
             info = "NA"  # When IP doesn't exist in the db, set info as "NA - Not Available"
         return info
+
+    async def detect_crawler(self, stats, bots_owner):
+        for _, path in enumerate(stats['paths']):
+            if path['path'] == '/robots.txt':
+                return (1.0, 0.0)
+        if stats['requests_in_second'] > 10:
+            if stats['user_agent'] in bots_owner:
+                return (0.85, 0.15)
+            return (0.5, 0.85)
+        if stats['user_agent'] in bots_owner:
+            hostname, _, _ = await self._loop.run_in_executor(
+                None, socket.gethostbyaddr, stats['peer_ip']
+            )
+            if 'search.msn.com' or 'googlebot.com' in hostname:
```

Comment:
can we extend this list?

Reply:
I think we can create a new PR, in which we can add many new bot owners and host names, also remove any outdated bots. What do you think about this?